testtools/runtest.py

```diff
@@ -76,9 +83,17 @@ def run(self, result=None):
             actual_result.startTestRun()
         else:
             actual_result = result
+        if self._pr:
+            self._pr.enable()
         try:
             return self._run_one(actual_result)
         finally:
+            if self._pr:
+                self._pr.disable()
+                self._pr.dump_stats("%s.%s.%s.cprof"
+                                    % (self.case.__class__.__module__,
+                                       self.case.__class__.__name__,
+                                       self.case._get_test_method().__name__))
```

Comment:
@jml no, it prints it out to a file. But attaching to a result sounds like a good idea.